app/views/api/v3/analytics/user_analytics/show_v2.html.erb

```diff
@@ -212,16 +209,16 @@
                }
     %>
 
-    <%= render partial: "api/v3/analytics/user_analytics/diagnosis_report",
-               locals: {
-                 page_id: "hypertension-report",
-                 report_diagnosis: t("progress_tab.diagnoses.hypertension"),
-                 title: t("progress_tab.hypertension_report_title"),
-                 subtitle: t("progress_tab.last_updated_at",
-                             date: display_date(@user_analytics.last_updated_at),
-                             time:  display_time(@user_analytics.last_updated_at)),
-                 diagnosis_data: @hypertension_reports_data
-               }
+    <%= render(ProgressTab::Hypertension::DiagnosisReportComponent.new(
+      service: @service,
+      current_facility: @current_facility,
+      current_user: @current_user,
+      period: @period,
+      drug_stock_query: @drug_stock_query,
+      drugs_by_category: @drugs_by_category,
+      last_updated_at: @user_analytics.last_updated_at,
+      cohort_data: @user_analytics.statistics.dig(:cohorts)
+    ))
     %>
 
     <%= render partial: "api/v3/analytics/user_analytics/diagnosis_report",
```

Comment:
Let's leave a todo here to refactor this into a view component and remove the partial?

Reply:
Sure!